polyfills/Array/prototype/indexOf/polyfill.js

```diff
@@ -1,20 +1,50 @@
-Array.prototype.indexOf = function indexOf(searchElement) {
-	if (this === undefined || this === null) {
-		throw new TypeError(this + ' is not an object');
+// 22.1.3.12. Array.prototype.indexOf ( searchElement [ , fromIndex ] )
+CreateMethodProperty(Array.prototype, 'indexOf', function indexOf(searchElement /* [ , fromIndex ] */) {
+	// 1. Let O be ? ToObject(this value).
+	var O = ToObject(this);
+	// 2. Let len be ? ToLength(? Get(O, "length")).
+	var len = ToLength(Get(O, "length"));
+	// 3. If len is 0, return -1.
+	if (len === 0) {
+		return -1;
 	}
-
-	var
-	arraylike = this instanceof String ? this.split('') : this,
-	length = Math.max(Math.min(arraylike.length, 9007199254740991), 0) || 0,
-	index = Number(arguments[1]) || 0;
-
-	index = (index < 0 ? Math.max(length + index, 0) : index) - 1;
-
-	while (++index < length) {
-		if (index in arraylike && arraylike[index] === searchElement) {
-			return index;
+	// 4. Let n be ? ToInteger(fromIndex). (If fromIndex is undefined, this step produces the value 0.)
+	var n = ToInteger(arguments[1]);
+	// 5. If n ≥ len, return -1.
+	if (n >= len) {
+		return -1;
+	}
+	// 6. If n ≥ 0, then
+	if (n >= 0) {
+		// a. If n is -0, let k be +0; else let k be n.
+		var k = n === -0 ? 0 : n;
```

Comment:
In cases like this (where `var k` appears in both the `if` and `else`), should the declaration be moved out?

Reply:
It could be moved out, do you think it should be moved out? 
As the variable declaration is done with `var`, it gets hoisted to the top of the function scope. This double declaration would get compiled away when users request the minified bundle.